src/ui/React/LinearForwardProgress.tsx

```diff
@@ -8,54 +8,52 @@ interface LinearForwardProgressProps {
   barColor?: React.CSSProperties["color"];
 }
 
+// Round percentage to nearest 5% interval to limit keyframe definitions
+const roundToInterval = (value: number, interval = 5): number => {
+  return Math.round(value / interval) * interval;
+};
+
+// Generate predefined keyframes for forward animations
+const generateKeyframes = (startPercent: number, endPercent: number) => {
+  const startPos = startPercent - 100;
+  const endPos = endPercent - 100;
+
+  return {
+    "0%": { transform: `translateX(${startPos}%)` },
+    "50%": { transform: "translateX(0%)" },
+    "50.01%": { transform: "translateX(-100%)" },
+    "100%": { transform: `translateX(${endPos}%)` },
+  };
+};
+
 export const LinearForwardProgress = React.forwardRef<unknown, LinearForwardProgressProps>(
   function LinearForwardProgress({ value, backgroundColor, barColor, ...otherProps }, ref): React.ReactElement {
     const previousValueRef = useRef(value);
-    const animationKeyRef = useRef(0);
 
     const isDecreasing = value < previousValueRef.current;
     const previousValue = previousValueRef.current;
 
     // Update ref for next render
     previousValueRef.current = value;
 
+    let animationStyles = {};
+
     if (isDecreasing) {
-      // Increment animation key to force new CSS animation
-      animationKeyRef.current += 1;
+      // Round values to 5% intervals to limit keyframe definitions
+      const roundedStartPercent = roundToInterval(previousValue);
```

Comment:
You might want to vary the rounding direction to prevent backwards motion. 

For instance, if endPercent is 5.1, that will get rounded up to 10, and then the next update might go backwards. The same thing can happen with startPervent.